dev-proxy-plugins/Inspection/DevToolsPlugin.cs

```diff
@@ -61,32 +68,67 @@ private static int GetFreePort()
+
         return configuration.PreferredBrowser switch
         {
-            PreferredBrowser.Chrome => RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
-                ? Environment.ExpandEnvironmentVariables(@"%ProgramFiles%\Google\Chrome\Application\chrome.exe")
-                : "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
-            PreferredBrowser.EdgeDev => RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
-                ? Environment.ExpandEnvironmentVariables(@"%ProgramFiles(x86)%\Microsoft\Edge Dev\Application\msedge.exe")
-                : "/Applications/Microsoft Edge Dev.app/Contents/MacOS/Microsoft Edge Dev",
-            _ => RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
-                ? Environment.ExpandEnvironmentVariables(@"%ProgramFiles(x86)%\Microsoft\Edge\Application\msedge.exe")
-                : "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge"
+            PreferredBrowser.Chrome => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Environment.ExpandEnvironmentVariables(@"%ProgramFiles%\Google\Chrome\Application\chrome.exe")
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "/usr/bin/google-chrome"
+                : throw new NotSupportedException("Unsupported operating system."),
+            PreferredBrowser.EdgeDev => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Environment.ExpandEnvironmentVariables(@"%ProgramFiles(x86)%\Microsoft\Edge Dev\Application\msedge.exe")
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "/Applications/Microsoft Edge Dev.app/Contents/MacOS/Microsoft Edge Dev"
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "/usr/bin/microsoft-edge-dev"
+                : throw new NotSupportedException("Unsupported operating system."),
+            _ => throw new NotSupportedException($"{configuration.PreferredBrowser} is an unsupported browser. Please change your PreferredBrowser setting for {Name}.")
         };
     }
 
     private void InitInspector()
     {
+        string processName = string.Empty;
+
+        try
+        {
+            processName = GetChromiumProcessName(_configuration);
+        }
+        catch (NotSupportedException ex)
+        {
+            Logger.LogError(ex, "Error starting {name}. Error finding the browser.", Name);
+        }
+
+        // check if the browser is installed
+        if (!File.Exists(processName))
+        {
+            Logger.LogError("Error starting {name}. Browser executable not found at {processName}", Name, processName);
+            return;
+        }
+
+        // find if the process is already running
+        var processes = Process.GetProcessesByName(Path.GetFileNameWithoutExtension(processName));
```

Comment:
This doesn't work on macOS because the actual process name is stored in `Process.MainModule.ModuleName`, so we need to update this logic to differentiate across OSes

Reply:
ok, I'll dust off my MPro and check.

Reply:
All good, I've done a couple of updates already